aptos-move/block-executor/src/executor.rs

```diff
@@ -143,6 +162,100 @@ where
+        versioned_cache: &MVHashMap<T::Key, T::Tag, T::Value, DelayedFieldID>,
+        executor: &E,
+        base_view: &S,
+        global_module_cache: &GlobalModuleCache<
+            ModuleId,
+            CompiledModule,
+            Module,
+            AptosModuleExtension,
+        >,
+        runtime_environment: &RuntimeEnvironment,
+        parallel_state: ParallelState<T>,
+        scheduler: &SchedulerV2,
+    ) -> Result<(), PanicError> {
+        let _timer = TASK_EXECUTE_SECONDS.start_timer();
+
+        // TODO(BlockSTMv2): proper integration w. execution pooling for performance.
+        let txn = signature_verified_block.get_txn(idx_to_execute);
+
+        let mut abort_manager = AbortManager::new(idx_to_execute, incarnation, scheduler);
+        let sync_view = LatestView::new(
+            base_view,
+            global_module_cache,
+            runtime_environment,
+            ViewState::Sync(parallel_state),
+            idx_to_execute,
+        );
+        let execution_result = executor.execute_transaction(&sync_view, txn, idx_to_execute);
+
+        let mut prev_modified_keys = last_input_output
+            .modified_keys(idx_to_execute, true)
+            .map_or_else(HashMap::new, |keys| keys.collect());
+        let mut read_set = sync_view.take_parallel_reads();
+        if read_set.is_incorrect_use() {
+            return Err(code_invariant_error(format!(
+                "Incorrect use detected in CapturedReads after executing txn = {idx_to_execute} incarnation = {incarnation}"
+            )));
+        }
+
+        let maybe_output =
+            Self::process_execution_result(&execution_result, &mut read_set, idx_to_execute)?;
+
+        // TODO: BlockSTMv2: use estimates for delayed field reads? (see V1 update on abort).
+        let mut resource_write_set = Vec::new();
+        if let Some(output) = maybe_output {
+            resource_write_set = output.resource_write_set();
+            for (key, value, maybe_layout) in resource_write_set.clone().into_iter() {
+                remove_from_previous_keys::<T>(&mut prev_modified_keys, &key, KeyKind::Resource)?;
+
+                abort_manager.invalidate_dependencies(versioned_cache.data().write_v2::<false>(
+                    key,
+                    idx_to_execute,
+                    incarnation,
+                    value,
+                    maybe_layout,
+                ))?;
+            }
+            // TODO(BlockSTMv2): handle groups, delayed fields and aggregator v1.
+        }
+
+        // Remove entries from previous write/delta set that were not overwritten.
+        for (key, kind) in prev_modified_keys {
+            use KeyKind::*;
+            match kind {
+                Resource => {
+                    abort_manager.invalidate_dependencies(
+                        versioned_cache
+                            .data()
+                            .remove_v2::<false>(&key, idx_to_execute),
```

Comment:
nit: maybe we can wrap these `::<true/false>` as functions to be `remove_group_metadata_v2(...)` vs `remove_...`?

Reply:
I have a TODO to refactor MVDataMap so we don't have to create fake WriteOps for metadata. It would also allow storing versioned sizes alongside instead of having a different re-implementation in MVGroupMap. I was thinking we can do proper interfaces when we address the TODO since probably those will change anyway.